ntbs-service-unit-tests/DataMigration/ImportValidatorTest.cs

```diff
@@ -127,26 +127,463 @@ public async Task ImportValidatorValidatesCollectionModels()
+                AdultsActiveTB = 2,
+                AdultsLatentTB = 2,
+                AdultsStartedTreatment = 1,
+                AdultsFinishedTreatment = 1,
+            };
+
+            // Act
+            var validationResults = await _importValidator.CleanAndValidateNotification(null, RunId, notification);
+
+            // Assert
+            // Validation errors
+            var errorMessages = validationResults.Select(r => r.ErrorMessage).ToList();
+            Assert.Empty(errorMessages);
+
+            // Data cleaning warnings
+            _importLoggerMock.Verify(s => s.LogNotificationWarning(null, RunId, notification.LegacyId,
+                ValidationMessages.ContactTracingAdultsActiveTB));
+
+            Assert.Equal(3, notification.ContactTracing.AdultsIdentified);
+            Assert.Equal(3, notification.ContactTracing.AdultsScreened);
+            Assert.Null(notification.ContactTracing.AdultsActiveTB);
+            Assert.Equal(2, notification.ContactTracing.AdultsLatentTB);
+            Assert.Equal(1, notification.ContactTracing.AdultsStartedTreatment);
+            Assert.Equal(1, notification.ContactTracing.AdultsFinishedTreatment);
+        }
+
         [Fact]
-        public async Task ImportValidatorCleansContactTracingValues()
+        public async Task ImportValidatorCleansContactTracingValues_WhenTooManyOfEachChildrenCaseType()
         {
             // Arrange
             var notification = CreateBasicNotification();
 
-            // These values would fail validation
             notification.ContactTracing = new ContactTracing
             {
-                AdultsIdentified = 1,
-                AdultsScreened = 3
+                ChildrenIdentified = 3,
+                ChildrenScreened = 3,
+                ChildrenActiveTB = 2,
+                ChildrenLatentTB = 2,
+                ChildrenStartedTreatment = 1,
+                ChildrenFinishedTreatment = 1,
             };
 
             // Act
             var validationResults = await _importValidator.CleanAndValidateNotification(null, RunId, notification);
 
             // Assert
+            // Validation errors
             var errorMessages = validationResults.Select(r => r.ErrorMessage).ToList();
+            Assert.Empty(errorMessages);
+
+            // Data cleaning warnings
+            _importLoggerMock.Verify(s => s.LogNotificationWarning(null, RunId, notification.LegacyId,
+                ValidationMessages.ContactTracingChildrenActiveTB));
+
+            Assert.Equal(3, notification.ContactTracing.ChildrenIdentified);
+            Assert.Equal(3, notification.ContactTracing.ChildrenScreened);
+            Assert.Null(notification.ContactTracing.ChildrenActiveTB);
+            Assert.Equal(2, notification.ContactTracing.ChildrenLatentTB);
+            Assert.Equal(1, notification.ContactTracing.ChildrenStartedTreatment);
+            Assert.Equal(1, notification.ContactTracing.ChildrenFinishedTreatment);
+        }
```

Comment:
Finally, I'm unsure whether this is the desired behaviour or not, but it is a reasonable one I think. I've messaged Nancy to confirm the requirements

Reply:
Oh sorry, just seen this comment, which I guess is equivalent to my one above!